src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupBasedAlleles.java

```diff
@@ -0,0 +1,105 @@
+package org.broadinstitute.hellbender.utils.pileup;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.VariantContextBuilder;
+import org.broadinstitute.hellbender.engine.*;
+import org.broadinstitute.hellbender.tools.walkers.mutect.AlignmentData;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import scala.Char;
+
+import java.util.*;
+
+
+public final class PileupBasedAlleles {
+
+
+    ArrayList<VariantContext> forcedPileupAlleles;
+
+
+    public static ArrayList<VariantContext> getPileupVariantContexts(List<AlignmentData> alignmentDataList) {
+
+        ArrayList<VariantContext> pileupSNPsList = new ArrayList<>();
+
+        for(AlignmentData alignmentData : alignmentDataList) {
+            AlignmentContext alignmentContext = alignmentData.getAlignmentContext();
+            ReferenceContext referenceContext = alignmentData.getReferenceContext();
+            final int numOfBases = alignmentContext.size();
+            final ReadPileup pileup = alignmentContext.getBasePileup();
+            final byte refBase = referenceContext.getBase();
+
+            Map<String, Integer> insertionCounts = new HashMap<>();
+
+            Map<Byte, Integer> altCounts = new HashMap<>();
+
+            for (PileupElement element : pileup) {
+//            for (byte eachBase : pileup.getBases()) {
+                byte eachBase = element.getBase();
+                // check to see that the base is not ref and that the alleles are one of these bases - ATCGN
+//                if (refBase != eachBase && Allele.acceptableAlleleBases(new byte[eachBase])) {
+                // TODO: AH & BG add a better check for acceptable alleles for eachBase
+                if (refBase != eachBase && eachBase != 68) {
+                    incrementAltCount(eachBase, altCounts);
+                }
+
+                // now look for indels
+                if (element.isBeforeInsertion()) {
```

Comment:
@jamesemery unclear what you mean here.

Reply:
There are some other projects that are interested in this code. They are not necessarily interested in the pileup indel calling code right now (since its complicated and they might have their own versions of this code) Ideally there should be a toggle that disables all of the indel pileup code. (or that disables putting them into the haplotypes).